include/NetworkDevice.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "Packet.hpp"
+
+#include <vector>
+#include <memory>
+#include <cstdint>
+#include <unordered_map>
+
+enum class DeviceType: std::uint8_t {
+    Switch,
+    ComputerSenderNode,
+    ComputerReceiverNode,
+};
+
+class NetworkDevice {
+public:
+    explicit NetworkDevice(DeviceType type, std::uint32_t id);
+    virtual ~NetworkDevice() = default;
+
+    /**
+     * @brief Vector of pointers to adjacent network devices.
+     *
+     * @details Usage of std::weak_ptr is necessary to avoid cyclic links.
+     */
+    std::vector<std::weak_ptr<NetworkDevice>> adjacent_devices;
+
+    /**
+     * @brief Network device type.
+     */
+    DeviceType type;
+
+    /**
+     * @brief Network device ID.
+     */
+    std::uint32_t id;
```

Comment:
Do we really need an `id` or passing pointers to the objects is sufficient?

Reply:
The `id` is currently being used by the `storage`, but without last one we can use pointers to objects instead.